test/contrib/dropbox_test.py

```diff
@@ -0,0 +1,284 @@
+        self.assertTrue(DROPBOX_TEST_SIMPLE_FILE in list)  # we verify recursivity
+
+    def test_listdir_multiple(self):
+        list = self.luigiconn.listdir(DROPBOX_TEST_PATH, limit=2)
+        self.assertTrue('/' not in list)
+        self.assertTrue(DROPBOX_TEST_PATH in list)
+        self.assertTrue(DROPBOX_TEST_SIMPLE_FILE in list)  # we verify recursivity
+
+    def test_remove(self):
+        self.assertTrue(self.luigiconn.exists(DROPBOX_TEST_FILE_TO_DELETE_1))
+        self.assertTrue(self.luigiconn.exists(DROPBOX_TEST_FILE_TO_DELETE_2))
+        self.luigiconn.remove(DROPBOX_TEST_FILE_TO_DELETE_1)
+        self.assertFalse(self.luigiconn.exists(DROPBOX_TEST_FILE_TO_DELETE_1))
+        self.assertTrue(self.luigiconn.exists(DROPBOX_TEST_FILE_TO_DELETE_2))
+
+        self.luigiconn.remove(DROPBOX_TEST_DIR_TO_DELETE)
+        self.assertFalse(self.luigiconn.exists(DROPBOX_TEST_FILE_TO_DELETE_2))
+
+    def test_remove_nonexisting(self):
+        self.assertFalse(self.luigiconn.remove(DROPBOX_TEST_NON_EXISTING_FILE))
+
+    def test_mkdir_new_dir(self):
+        self.assertFalse(self.luigiconn.exists(DROPBOX_TEST_DIR_TO_CREATE))
+        self.assertFalse(self.luigiconn.exists(DROPBOX_TEST_OUTER_DIR_TO_CREATE))
+        self.luigiconn.mkdir(DROPBOX_TEST_DIR_TO_CREATE)
+        self.assertTrue(self.luigiconn.isdir(DROPBOX_TEST_OUTER_DIR_TO_CREATE))
+        self.assertTrue(self.luigiconn.isdir(DROPBOX_TEST_DIR_TO_CREATE))
+
+    def test_mkdir_recreate_dir(self):
+        try:
+            self.luigiconn.mkdir(DROPBOX_TEST_SIMPLE_DIR)
+        except Exception as ex:
+            self.fail("mkdir with default options raises Exception:" + str(ex))
+
+        try:
+            self.luigiconn.mkdir(DROPBOX_TEST_SIMPLE_DIR, raise_if_exists=False)
+        except Exception as ex:
+            self.fail("mkdir with 'raise_if_exists=False' raises Exception:" + str(ex))
+
+        with self.assertRaises(luigi.target.FileAlreadyExists):
+            self.luigiconn.mkdir(DROPBOX_TEST_SIMPLE_DIR, raise_if_exists=True)
+
+    def test_mkdir_recreate_file(self):
+        with self.assertRaises(luigi.target.NotADirectory):
+            self.luigiconn.mkdir(DROPBOX_TEST_SIMPLE_FILE)
+
+        with self.assertRaises(luigi.target.NotADirectory, ):
+            self.luigiconn.mkdir(DROPBOX_TEST_SIMPLE_FILE, raise_if_exists=True)
+
+        with self.assertRaises(luigi.target.NotADirectory, ):
+            self.luigiconn.mkdir(DROPBOX_TEST_SIMPLE_FILE, raise_if_exists=False)
+
+    def test_isdir(self):
+        self.assertTrue(self.luigiconn.isdir('/'))
+        self.assertTrue(self.luigiconn.isdir(DROPBOX_TEST_PATH))
+        self.assertTrue(self.luigiconn.isdir(DROPBOX_TEST_SIMPLE_DIR))
+        self.assertFalse(self.luigiconn.isdir(DROPBOX_TEST_SIMPLE_FILE))
+        self.assertFalse(self.luigiconn.isdir(DROPBOX_TEST_NON_EXISTING_FILE))
+
+    def test_move(self):
+        md, res = self.dropbox_api.files_download(DROPBOX_TEST_FILE_TO_MOVE_ORIG)
```

Comment:
did you mean to explicitly use the direct dropbox method over your `download_as_byte()` method?

Reply:
Yes, I am intentionally using the download_as_byte() method from the official dropbox api.

I want to check that the contents of the file are still the same after moving the file.

Since I only want to test my 'move' method (and not any other of my methods), I decided to get the contents of the file using the official api instead using of my download method.

Reply:
Thoughts on testing that your download_byte() method passes tests and then using it elsewhere? i'm not set on telling you to use one over the other. but download_bytes does hide the logic of returning just the content from the api response